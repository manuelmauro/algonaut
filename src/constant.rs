/// Application ID prefix when signing
pub const APPID_PREFIX: &[u8; 5] = b"appID";

/// how long addresses are in bytes
pub const KEN_LEN_BYTES: u64 = 32;
