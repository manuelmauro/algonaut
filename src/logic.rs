use crate::constant::APPID_PREFIX;
use sha2::{Digest, Sha512};

pub fn get_application_address(app_id: u64) -> Vec<u8> {
    let app_id: &[u8; 8] = &app_id.to_be_bytes();
    let to_sign: Vec<u8> = APPID_PREFIX
        .iter()
        .copied()
        .chain(app_id.iter().copied())
        .collect();

    let mut hasher = Sha512::new();
    hasher.update(to_sign);

    hasher.finalize()[..].to_vec()
}

#[cfg(test)]
mod tests {
    use super::get_application_address;

    #[test]
    fn get_application_address_snapshots() {
        let a = get_application_address(13);
        insta::assert_yaml_snapshot!(a);
    }
}
