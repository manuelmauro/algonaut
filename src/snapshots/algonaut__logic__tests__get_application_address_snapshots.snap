---
source: src/logic.rs
expression: a
---
- 174
- 108
- 145
- 15
- 43
- 99
- 36
- 10
- 139
- 82
- 201
- 202
- 142
- 242
- 99
- 74
- 61
- 193
- 187
- 44
- 151
- 217
- 248
- 206
- 180
- 209
- 90
- 215
- 232
- 66
- 175
- 204
- 169
- 225
- 82
- 128
- 30
- 94
- 110
- 70
- 3
- 59
- 48
- 108
- 176
- 169
- 159
- 62
- 187
- 88
- 65
- 125
- 161
- 12
- 43
- 23
- 172
- 246
- 74
- 228
- 121
- 195
- 177
- 111

